astropy/coordinates/representation.py

```diff
@@ -1902,6 +1947,27 @@ def from_cartesian(cls, cart):
         # erfa p2s: P-vector to spherical polar coordinates.
         return cls(*erfa_ufunc.p2s(p), copy=False)
 
+    def transform(self, matrix):
+        r"""Transform the spherical coordinates using a 3x3 matrix.
+
+        This returns a new representation and does not modify the original one.
+        Any differentials attached to this representation will also be
+        transformed.
+
+        Parameters
+        ----------
+        matrix : (3,3) `~numpy.ndarray`
+            A 3x3 transformation matrix, such as a rotation matrix.
+
+        """
+        # TODO! differentials
+        xyz = erfa_ufunc.s2c(self.lon, self.lat)
+        p = erfa_ufunc.rxp(matrix, xyz)
+        lon, lat, ur = erfa_ufunc.p2s(p)
+        rep = self.__class__(lon=lon, lat=lat, distance=ur * self.distance)
+
+        return rep
```

Comment:
I'd just `return self.__class__(...)`

Reply:
I can certainly change it, but I had 2 (currently) bad reasons for writing it as-is.
1. for when the separate case of differentials is handled
2. static typers don't like returning an object that isn't bound to a variable as they have difficulty inspecting the type. So if this function were annotated as a `matrix: np.array) -> SphericalRepresentation:`, then `mypy` would complain, but not as it is currently written.

So... I'll change it.